src/Abstractions/TaskOptions.cs

```diff
@@ -100,4 +103,7 @@ public SubOrchestrationOptions(TaskOptions options, string? instanceId = null)
 /// The time when the orchestration instance should start executing. If not specified or if a date-time in the past
 /// is specified, the orchestration instance will be scheduled immediately.
 /// </param>
-public record StartOrchestrationOptions(string? InstanceId = null, DateTimeOffset? StartAt = null);
+/// <param name="OrchestrationIdReusePolicy">The orchestration reuse policy. This allows for the reuse of an instance ID
+/// as well as the options for it.</param>
+public record StartOrchestrationOptions(string? InstanceId = null, DateTimeOffset? StartAt = null,
+ Dictionary<List<OrchestrationOptions.OrchestrationRuntimeStatus>, OrchestrationOptions.InstanceIdReuseAction>? OrchestrationIdReusePolicy = null);
```

Comment:
I see what you are trying to solve here: making it more granular on what the behavior is based on the state of the orchestration. Which I do think is worth considering.

I see the following problems with this approach:

1. This information does not align with the protobuf spec
2. It is a bit unwieldy for customers to need to specify this dictionary.

Instead, here is what I recommend:

1. Lets first evaluate if this is needed? Or can we decide and document what the behavior of the 3 individual policies are based on the orchestration state? For example, we may consider applying conflict resolution for non-terminal orchestrations only.
2. If not, then we can expand the protobuf and our enum to have more states that better clarify what happens. IE: `ErrorAlways`, `ErrorIfRunning` eg.

The end result should be customers need to only provide a single enum to this options object.